doc/man3/EVP_PKEY_get_default_digest_nid.pod

```diff
@@ -2,31 +2,44 @@
 
 =head1 NAME
 
-EVP_PKEY_get_default_digest_nid - get default signature digest
+EVP_PKEY_get_default_digest_nid, EVP_PKEY_get_default_digest_name
+- get default signature digest
 
 =head1 SYNOPSIS
 
  #include <openssl/evp.h>
+
+ int EVP_PKEY_get_default_digest_name(EVP_PKEY *pkey,
+                                      char *mdname, size_t mdname_sz)
  int EVP_PKEY_get_default_digest_nid(EVP_PKEY *pkey, int *pnid);
 
 =head1 DESCRIPTION
 
-The EVP_PKEY_get_default_digest_nid() function sets B<pnid> to the default
-message digest NID for the public key signature operations associated with key
-B<pkey>. Note that some signature algorithms (i.e. Ed25519 and Ed448) do not use
-a digest during signing. In this case B<pnid> will be set to NID_undef.
+EVP_PKEY_get_default_digest_name() fills in the default message digest
+name for the public key signature operations associated with key
+I<pkey> into I<mdname>, up to at most I<mdname_sz> bytes including the
+ending NUL byte.
+
+EVP_PKEY_get_default_digest_nid() sets I<pnid> to the default message
+digest NID for the public key signature operations associated with key
+I<pkey>.  Note that some signature algorithms (i.e. Ed25519 and Ed448)
+do not use a digest during signing.  In this case I<pnid> will be set
+to NID_undef.  This function is only reliable for legacy keys, which
+is keys loaded from engines, or created with internal libcrypto
+implementations (B<EVP_PKEY_ASN1_METHOD>).
```

Comment:
I'm not sure I understood what you meant by "created with internal libcrypto implementations"

Reply:
Does the type **EVP_PKEY_ASN1_METHOD** help?  I'm all ears for a better term

Reply:
Oh - I see what you're getting at. Perhaps:

"This function is only reliable for legacy keys, which are keys loaded from engines, or created via EVP_PKEY_assign() or have a custom EVP_PKEY_ASN1_METHOD"

Reply:
I reworded it a bit further, I hope you're pleased with the result.  Your input was very helpful to get there, thanks!